main_test.go

```diff
@@ -7,35 +7,74 @@ import (
 
 func TestParseTextToJSON(t *testing.T) {
 	tests := []struct {
-		name     string
-		input    string
-		expected map[string]interface{}
+		name           string
+		input          string
+		expectedOutput interface{}
+		expectedError  error
 	}{
 		{
 			name:  "Test simple parsing",
 			input: `name:"Fikar" age:10 address:{city:"Jakarta Selatan" state:"DKI Jakarta"} gender:"male" is_married:false`,
-			expected: map[string]interface{}{
-				"name": "Fikar",
-				"age":  10,
-				"address": map[string]interface{}{
-					"city":  "Jakarta Selatan",
-					"state": "DKI Jakarta",
-				},
+			expectedOutput: map[string]interface{}{
+				"name":       "Fikar",
+				"age":        10,
+				"address":    map[string]interface{}{"city": "Jakarta Selatan", "state": "DKI Jakarta"},
 				"gender":     "male",
 				"is_married": false,
 			},
+			expectedError: nil,
+		},
+		{
+			name:  "Test scientific notation number",
+			input: `name:"John" age:30 salary:2e+06 married:true`,
+			expectedOutput: map[string]interface{}{
+				"name":    "John",
+				"age":     30,
+				"salary":  2000000.0,
```

Comment:
```suggestion
				"salary":  2e+06,
```
i think we should retain user's input as it is, wdyt?

Reply:
fair enough. updated the code